### ISSUE TITLE
[Bug] Sofle `via` keymap Missing RGB Underglow Animations for Sofle RGB 2.1

### ISSUE THREAD

**Issue by joshdick** [qmk/qmk_firmware#23251]
### Describe the Bug

**tl;dr: Is it possible that #22934 removed `RGBLIGHT_ENABLE` in favor of `RGB_MATRIX_ENABLE`, which in turn disabled RGB underglow animations for `-kb sofle/rev1 -km via` that were previously working?**

---

I have a Sofle RGB 2.1 (standard/not the Keyhive version.) My keyboard uses USB-C Sea-Micros and has 6 underglow LEDs per side, but no per-key LEDs. 

Several months back, I had previously built firmware with no local modifications, using the `via` keymap for VIA compatibility (`qmk compile -kb sofle/rev1 -km via`) and had been using it without experiencing the issue described below.

However, after updating my QMK version/repository and rebuilding the firmware using the command above, RGB underglow animations no longer work, and the RGB mode cycling keys I've assigned in VIA  (`RGB_MOD`/`RGB_RMOD`)  have no effect. I can still toggle the lights on and off, cycle through static hues (`RGB_HUI`/`RGB_HUD`), and modify brightness/saturation; it's only the animations that no longer work. I also noticed that VIA reports backlight-related errors when connecting to the keyboard using that firmware built wth `-km via`, with references to `Command name: BACKLIGHT_CONFIG_GET_VALUE` in its error log.

My guess is that this behavior may have been caused by #22934 by @waffle87, which included [a change to the `via` keymap `rules.mk`](https://github.com/qmk/qmk_firmware/pull/22934/files#diff-8ee57947884f2303bdea5b4086c2331429c84cd2388b18e5951081a669a301ac) that seemingly disabled `RGBLIGHT_ENABLE` in favor of `RGB_MATRIX_ENABLE,` along with [previously-defined `RGBLIGHT_EFFECT`s](https://github.com/qmk/qmk_firmware/pull/22934/files#diff-67ad564681c29ef57964fd6131cc51b56e9ed8cceae7314a7688b9612c54e00cL30-L48), but it's possible I'm misunderstanding those changes.

I tried modifying the `via` keymap `rules.mk` locally to add `RGBLIGHT_ENABLE = yes` and either remove `RGB_MATRIX_ENABLE = yes` or set it to `no`, but both of these had no effect on the firmware size and neither were able get the animations working as before on their own. In combination with `RGBLIGHT_ENABLE = yes` and `RGB_MATRIX_ENABLE = no`, I added a custom `config.h` that included several of the `#define RGBLIGHT_EFFECT_`s that I linked above, and once included, **those work as expected and the errors disappear in VIA**, but I get "firmware too large" errors if I try to include more than ~3 effects whereas in the previously-working firmware, I believe many more were available.

So, to repeat my opening question: Is it possible that #22934 removed `RGBLIGHT_ENABLE` in favor of `RGB_MATRIX_ENABLE`, which in turn disabled RGB underglow animations for `-kb sofle/rev1 -km via` that were previously working?

### Keyboard Used

sofle/rev1

### Operating System

macOS

### qmk doctor Output

```text
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.1.2
Ψ QMK home: /Users/joshdick/qmk_firmware
Ψ Detected macOS 14.4 (Apple Silicon).
Ψ Userspace enabled: False
Ψ Git branch: master
Ψ Repo version: 0.24.2
Ψ - Latest master: 2024-03-08 14:44:32 -0500 (62e2cb9286) -- [Keyboard] Add sf2040 (#23211)
Ψ - Latest upstream/master: 2024-03-09 21:40:19 +1100 (900e79febe) -- Attempt to deal with GCC 13+ and rv32 arch extensions. (#23086)
Ψ - Latest upstream/develop: None
Ψ - Common ancestor with upstream/master: 2024-03-08 14:44:32 -0500 (62e2cb9286) -- [Keyboard] Add sf2040 (#23211)
Ψ - Common ancestor with upstream/develop: None
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 8.3.1
Ψ Found avr-gcc version 8.5.0
Ψ Found avrdude version 7.3
Ψ Found dfu-programmer version 1.1.0
Ψ Found dfu-util version 0.11
Ψ Submodules are up to date.
Ψ Submodule status:
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)
Ψ - lib/chibios-contrib: 2023-11-27 18:15:44 +0100 --  (9d7a7f90)
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)
Ψ QMK is ready to go
```

### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)
